plugins/reports/counter/CounterReportPlugin.inc.php

```diff
@@ -94,346 +207,72 @@ function setBreadcrumbs() {
-			$reportRelease = $reportRequestNode->getChildByName($sushiPrefix.'ReportDefinition')->getAttribute('Release');
-
-			$usageDateRange = $reportRequestNode->getChildByName($sushiPrefix.'ReportDefinition')->getChildByName($sushiPrefix.'Filters')->getChildByName($sushiPrefix.'UsageDateRange');
-			$usageDateBegin = $usageDateRange->getChildByName($sushiPrefix.'Begin')->getValue();
-			$usageDateEnd = $usageDateRange->getChildByName($sushiPrefix.'End')->getValue();
-
-
-			$this->_assignTemplateCounterXML($templateManager, $usageDateBegin, $usageDateEnd, $useLegacyStats);
-
-			$templateManager->assign('requestorID', $requestorID);
-			$templateManager->assign('requestorName', $requestorName);
-			$templateManager->assign('requestorEmail', $requestorEmail);
-			$templateManager->assign('customerReferenceID', $customerReferenceID);
-			$templateManager->assign('reportName', $reportName);
-			$templateManager->assign('reportRelease', $reportRelease);
-			$templateManager->assign('usageDateBegin', $usageDateBegin);
-			$templateManager->assign('usageDateEnd', $usageDateEnd);
-
-			$templateManager->assign('templatePath', $this->getTemplatePath());
-
-			$templateManager->display($this->getTemplatePath() . 'sushixml.tpl', 'text/xml');
-		}
-	}
-
-	/**
-	* Internal function to form some of the CSV columns
-	*/
-	function _formColumns(&$cols, $entries) {
-		$allMonthsTotal = 0;
-		$currTotal = 0;
-		$htmlTotal = 0;
-		$pdfTotal = 0;
-		for ($i = 1; $i <= 12; $i++) {
-			$currTotal = 0;
-			foreach ($entries as $entry) {
-				$month = (int) substr($entry[STATISTICS_DIMENSION_MONTH], 4, 2);
-				if ($i== $month) {
-					$metric = $entry[STATISTICS_METRIC];
-					$currTotal += $metric;
-					if ($entry[STATISTICS_DIMENSION_FILE_TYPE] == STATISTICS_FILE_TYPE_HTML) {
-						$htmlTotal += $metric;
-					} else if ($entry[STATISTICS_DIMENSION_FILE_TYPE] == STATISTICS_FILE_TYPE_PDF) {
-						$pdfTotal += $metric;
+					// Legacy COUNTER Release 3
+					import('plugins.reports.counter.classes.reports.JR1_r3_0');
+					$r3jr1 = new JR1_r3_0('reports', $this->getName());
+					$r3jr1->_display($request);
+					return;
+				case 'fetch':
+					// Modern COUNTER Releases
+					// must provide a release, report, and year parameter
+					if ($request->getUserVar('release') && $request->getUserVar('report') && $request->getUserVar('year')) {
+						// release, report and year parameters must be sane
+						if (isset($available[$request->getUserVar('release')][$request->getUserVar('report')]) && in_array($request->getUserVar('year'), $years)) {
+							// try to get the report
+							$reporter = $this->getReporter($request->getUserVar('report'), $request->getUserVar('release'));
+							if ($reporter) {
+								// default report parameters with a yearlong range
+								$xmlResult = $reporter->getMetricsXML(array(), array(STATISTICS_DIMENSION_MONTH => array('from' => $request->getUserVar('year').'01', 'to' => $request->getUserVar('year').'12')));
```

Comment:
I don't think these exist (at least not yet):
master: https://github.com/pkp/ojs/blob/09042fea007be16bc76d464434bdbd49566a1163/classes/statistics/MetricsDAO.inc.php#L110-L115

dev:https://github.com/pkp/ojs/blob/f400f097c2cb8618d6cff2efdb6d5a0e6bd55805/classes/statistics/MetricsDAO.inc.php#L122-L127

Think we should add them?